### PR TITLE
[tuner] Add pickle dump feature and update dispatch_kind detection method

### PR DESCRIPTION
**Pull request by RattataKing**
1. Use pickle to save the list of configurations in tune,py, and candidate_trackers from autotune.py
2. autotune.py load the configuration .pkl file and add to the candidate_trackers
3. Update dispatch_kind detection method to match the new changes in IREE

### CODE REVIEW

File: tuning/autotune.py [hunk 1]
```diff
@@ -441,7 +456,9 @@ def compile_candidates(
         candidate_trackers[index].compiled_vmfb_path = compiled_file
 
     handle_error(
-        condition=(good == 0), msg="Failed to compile all candidate .mlir files"
+        condition=(good == 0),
+        msg="Failed to compile all candidate .mlir files",
+        exit_program=False,
```
Comment (RattataKing) [inline tuning/autotune.py:461]:
```suggestion
        condition=(good == 0), msg="Failed to compile all candidate .mlir files"
```

File: tuning/tune.py [hunk 2]
```diff
@@ -800,6 +807,10 @@ def tune(
     else:
         assert False
 
+    with open(output / "configs.pkl", "wb") as file:
+        pickle.dump(configs, file)
+    tune_logger.INFO(f"Configurations .pkl is store in {output/'configs.pkl'}")
```
Comment (kuhar) [inline tuning/tune.py:812]:
```suggestion
    tune_logger.INFO(f"Configurations .pkl is stored in {output/'configs.pkl'}")
```

File: tuning/autotune.py [hunk 3]
```diff
@@ -511,7 +526,7 @@ def benchmark_top_candidates(
     )
 
     logging.critical(
-        f"Total: {len(benchmark_results)} | Benchmarked: {len(benchmarked_files)} | Failed: {len(benchmark_failed_files)} | Benchmarking Rate: {len(benchmarked_files)/(len(benchmark_results)*100):.1f}%"
+        f"Total: {len(benchmark_results)} | Benchmarked: {len(benchmarked_files)} | Failed: {len(benchmark_failed_files)} | Benchmarking Rate: {(len(benchmarked_files)/len(benchmark_results))*100}%"
```
Comment (kuhar) [inline tuning/autotune.py:529]: why should we change this?

File: tuning/tune.py [hunk 4]
```diff
@@ -651,13 +652,12 @@ def parse_mlir(mlir_text: str) -> ir.Module:
...
 ) -> ir.WalkResult:
-    if isinstance(op.opview, ireec.dialects._linalg_ops_gen.Conv2DNhwcHwcfOp):
+    if op.name == "linalg.conv_2d_nhwc_hwcf":
         walk_result.was_interrupted = True
         walk_result.dispatch_kind = DispatchKind.conv
 
         return ir.WalkResult.INTERRUPT
-
-    if isinstance(op.opview, ireec.dialects._util_ops_gen.FuncOp):
+    elif op.name == "util.func":
```
Comment (kuhar) [inline tuning/tune.py:660]: nit: no else after return, just `if`

File: tuning/autotune.py (hunk 3 above)
Comment (RattataKing) [reply tuning/autotune.py:529]: old division is `a / (b*100)`, should change to `(a/b) * 100`

File: tuning/tune.py (hunk 4 above)
Comment (RattataKing) [inline tuning/tune.py:660]:
```suggestion
    if op.name == "util.func":
```

**Review by kuhar — APPROVED** [review]

**Pull request merged**